### PR TITLE
Todos recording

### PR DESCRIPTION
**Pull request by urbanwide**

### CODE REVIEW

File: app/controllers/schools/interventions_controller.rb [hunk 1]
```diff
@@ -30,14 +30,25 @@ def edit
...
+        else
+          @intervention_type = @observation.intervention_type
+          render :new
+        end
       else
-        @intervention_type = @observation.intervention_type
-        render :new
+        @observation = @school.observations.new(observation_params.merge(observation_type: :intervention,
+                                                                         created_by: current_user))
+        authorize! :create, @observation
```
Comment (ldodds) [inline app/controllers/schools/interventions_controller.rb:44]: is this unnecessary with the new feature enabled?

File: app/services/activity_creator.rb [hunk 2]
```diff
@@ -31,13 +31,11 @@ def create_completed_audit_activities_observation
   end
 
   def create_activity_observation
-    points = @activity.activity_type.score_when_recorded_at(@activity.school, @activity.happened_on)
```
Comment (ldodds) [inline app/services/activity_creator.rb:34]: Was this intentional? Might have missed some other change, but with the FeatureFlag off doesn't this mean schools aren't scoring points?

Comment (urbanwide) [reply app/services/activity_creator.rb:34]: It's now in a call back on observation. Tidied it to be behave in the same way as activities

File: spec/support/context_extensions.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+module ContextExtensions
+  def context(description, *args, toggle_feature: nil, **metadata, &block)
```
Comment (ldodds) [inline spec/support/context_extensions.rb:2]: nice :)

File: app/controllers/schools/interventions_controller.rb (hunk 1 above)
Comment (urbanwide) [reply app/controllers/schools/interventions_controller.rb:46]: No, have put it back in. Thanks :-)

File: app/models/concerns/todos/assignable.rb [hunk 4]
```diff
@@ -12,9 +12,15 @@ module Assignable
 
       has_many :activity_type_tasks, through: :activity_type_todos, source: :task, source_type: 'ActivityType'
       has_many :intervention_type_tasks, through: :intervention_type_todos, source: :task, source_type: 'InterventionType'
+      # can't have a pure tasks relationship - as would return mixed objects of mixed types
 
       accepts_nested_attributes_for :activity_type_todos, allow_destroy: true
       accepts_nested_attributes_for :intervention_type_todos, allow_destroy: true
     end
+
```
Comment (ldodds) [inline app/models/concerns/todos/assignable.rb:20]: can be removed if not used?

File: app/models/concerns/todos/completable.rb [hunk 5]
```diff
@@ -5,10 +5,80 @@ module Completable
...
       has_many :completed_tasks, through: :completed_todos, source: :todo, class_name: 'Todo'
       has_many :completed_activity_types, through: :completed_todos, source: :activity_type
       has_many :completed_intervention_types, through: :completed_todos, source: :intervention_type
+
+      # used to restrict returned audits or programmes, only return those that should have
+      # tasks checked off against
+      scope :completable, -> { raise NoMethodError, 'Implement completable scope in subclass!' }
+    end
+
+    # def tasks(task_type = nil)
```
Comment (ldodds) [inline app/models/concerns/todos/completable.rb:18]: can be removed if unused?

**Review by ldodds — APPROVED** [review]
:+1:

**Pull request merged**